onnxruntime/python/tools/transformers/onnx_model_bert_tf.py

```diff
@@ -295,6 +295,112 @@ def process_embedding(self):
+            if v_nodes is None:
+                logger.debug("Failed to match v path")
+                continue
+            (transpose_v, reshape_v, add_v, matmul_v) = v_nodes
+            qk_nodes = self.match_parent_path(matmul_qkv, ['Softmax', 'Add', "Mul", 'MatMul'], [0, 0, 0, 0])
+            if qk_nodes is None:
+                logger.debug("Failed to match qk_paths")
+                continue
+            (softmax_qk, add_qk, mul_qk, matmul_qk) = qk_nodes
+            q_nodes = self.match_parent_path(matmul_qk, ['Transpose', 'Reshape', 'Add', 'MatMul'], [0, 0, 0, 0])
+            if q_nodes is None:
+                logger.debug("Failed to match q path")
+                continue
+            (transpose_q, reshape_q, add_q, matmul_q) = q_nodes
+            k_nodes = self.match_parent_path(matmul_qk, ['Transpose', 'Reshape', 'Add', 'MatMul'], [1, 0, 0, 0])
+            if k_nodes is None:
+                logger.debug("Failed to match k path")
+                continue
+            (transpose_k, reshape_k, add_k, matmul_k) = k_nodes
+
+            mask_nodes = self.match_parent_path(add_qk, ['Mul', 'Sub', 'Unsqueeze'], [1, 0, 1])
+            if mask_nodes is None:
+                logger.debug("Failed to match mask path")
+                continue
+            if not self.has_constant_input(mask_nodes[1], 1):
+                logger.debug("Sub node expected to have an input with constant value 1.0.")
+                continue
+
+            upper_mask_nodes = self.match_parent_path(mask_nodes[2], ['Reshape', 'Cast', 'Concat'], [0, 1, 0])
+            if upper_mask_nodes is None:
+                continue
+            mask_concat = upper_mask_nodes[2]
+            if len(mask_concat.input) == 3:
+                # Temporary work around[Tracy's model]: require 2-d mask input, the current model has a 3-d mask input
+                self.add_node(
+                    helper.make_node("Concat", [mask_concat.input[0], mask_concat.input[2]], [mask_concat.output[0]],
+                                     mask_concat.name + "_modified",
+                                     axis=0))
+                self.remove_node(mask_concat)
+
+            is_same_root = self.check_attention_input(matmul_q, matmul_k, matmul_v, parent, output_name_to_node)
+            if is_same_root:
+                mask_index = self.attention_mask.process_mask(mask_nodes[-1].input[0])
+                logger.debug("Create an Attention node.")
+                attention_node = self.attention_fusion.create_attention_node(mask_index, matmul_q, matmul_k, matmul_v,
+                                                                             add_q, add_k, add_v, parent.output[0],
+                                                                             reshape_qkv.output[0])
+                if parent.op_type == 'Reshape':
+                    # Temporary work around[Tracy's model]: we require the skiplayernorm and attention op be fed with 3-d input
```

Comment:
If the input to reshape node is 3d, we shall use that as input.

Reply:
for this model, the input dimension is 2-d